packages/grafana-ui/src/components/InteractiveTable/InteractiveTable.tsx

```diff
@@ -146,6 +146,10 @@ interface Props<TableData extends object> {
    * Render function for the expanded row. if not provided, the tables rows will not be expandable.
    */
   renderExpandedRow?: (row: TableData) => ReactNode;
+  /**
+   * Whether to show the "Expand all" button. Depends on renderExpandedRow to be provided. Defaults to false.
+   */
+  showExpandAll?: boolean;
```

Comment:
Seeing as this prop depends on another one, we can make the type support for that a bit better and provide a type error when `showExpandAll` is used without `renderExpandedRow`:

```tsx
interface WithExpandableRow<TableData extends object> extends BaseProps<TableData> {
  renderExpandedRow: (row: TableData) => ReactNode;
  showExpandAll?: boolean;
}

interface WithoutExpandableRow<TableData extends object> extends BaseProps<TableData> {
  renderExpandedRow?: never;
  showExpandAll?: never;
}

type Props<TableData extends object> = WithExpandableRow<TableData> | WithoutExpandableRow<TableData>;
```

Reply:
Nice!
![image](https://github.com/grafana/grafana/assets/113929542/c2e7b4ab-8969-4f7f-9815-8fb7a3245954)